### PR TITLE
Details attributes and http header encoding

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Since I'm working on a change in Brave, I figured I'd go ahead and elaborate the common attributes here. Further clarifications can either be made in this PR, or be saved for later ones.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,55 @@
...
+### Details
+
+It may not be obvious why you'd send Sampled=0 to the next hop. Imagine a service decides not to trace an operation and makes 2 out-going calls, and these branched out further. If 0 ("don't trace") isn't propagated, the system might receive only parts of the operation, confusing users.
+
+Leaving Sampled absent is special-case. The only known use-cases are the following:
+
+* Debug trace: When setting Flags to 1, sampling is implicit
+* Externally provisioned IDs: When you want to control IDs, but not sampling policy
+
+Unless it is a debug trace, leaving sampled unset is typically for ID correlation. For example, someone re-uses a global identifier from another system, or correlating in logs. In these cases, the caller knows the ID they want, but allows the next hop to decide if it will be traced or not. The caller should not report a span to the tracing system using this ID unless they propagate Sampled=1.
```
Comment (codefromthecrypt) [inline README.md:48]: cc @jcarres-mdsol @dangets This particularly relates to functionality in Finagle, which is missing in Brave https://github.com/openzipkin/brave/issues/277

File: README.md [hunk 2]
```diff
@@ -1,2 +1,55 @@
...
+
+### Http Header format
+The `X-B3-ParentSpanId` header must be present on a child span and absent on the root span. It is encoded as 16 hex characters. For example, a ParentSpanId header might look like: `X-B3-ParentSpanId: 0020000000000001`
+
+# Flags
+The following flags are reported either in a flag set or separate attributes.
+
+## Sampled
+
+When the Sampled is 1, report this span to the tracing system. When it is 0, do do not. When B3 attributes are sent without the Sampled attribute, the receiver should make the decision. Once Sampled is set to 0 or 1, the value should be consistently sent downstream.
```
Comment (shakuzen) [inline README.md]: "When it is 0, do do not." → "When it is 0, do not."

Comment (shakuzen) [inline README.md]: "When the Sampled is 1" sounds strange to me. Maybe "When Sampled is 1" or "When the Sampled attribute is 1"

**Review by shakuzen — APPROVED** [review]
This is easy to understand and should be a good guide for those using B3. 👍  I only left a few nit pick polish comments.

File: README.md (hunk 2 above)
Comment (codefromthecrypt) [reply README.md]: Thanks.. I think I meant `When the Sampled flag is 1`

**Comment by codefromthecrypt** [comment]
I'm going to merge this in the spirit of getting something far better than the template README out there. Will take any more comments after commit, so please voice if you have them!

**Pull request merged**